src/pages/net-neutrality/petition.js

```diff
@@ -41,38 +66,55 @@ var Signup = React.createClass({
     }
     this.stickyForm.onResize();
   },
+  generateModal: function() {
+    if (this.state.showModal) {
+      if (this.state.signupSuccess) {
+        return this.generateSignupModal();
+      }
+      return this.generateDonationModal();
+    }
+    return null;
+  },
+  generateSignupModal: function() {
+    return (
+      <Modal onClose={this.closeModal}>
```

Comment:
shouldn't this be `onClose={() => this.closeModal()}`?

Reply:
or is this a version of react that does the binding for us?

Reply:
copy-pasted. I've left it as is, mostly because we'd be hitting it anyway if we want to do a proper uplift. Since this code still uses `createClass` with an object rather than `class ... extends ...`, autobinding still happens.